flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1262,11 +1261,19 @@ public void stopCheckpointScheduler() {
 	 */
 	public void abortPendingCheckpoints(CheckpointException exception) {
 		synchronized (lock) {
-			for (PendingCheckpoint p : pendingCheckpoints.values()) {
-				failPendingCheckpoint(p, exception.getCheckpointFailureReason());
-			}
+			// do not traverse pendingCheckpoints directly, because it might be changed
+			abortPendingCheckpoints(
+				pendingCheckpoints.values().toArray(new PendingCheckpoint[0]), exception);
+		}
+	}
+
+	private void abortPendingCheckpoints(
+		PendingCheckpoint[] checkpoints, CheckpointException exception) {
```

Comment:
nit: if one parameter is indented, all of them should be (according to our code style)

Reply:
Alright, I need to re-check the code style guide :)